clients/python-wrapper/lakefs/client.py

```diff
@@ -106,6 +110,70 @@ def version(self) -> str:
         return self._server_conf.version
 
 
+def get_identity_token(session: 'boto3.Session') -> str:
+    """
+   Generate the identity token required for lakeFS authentication from an AWS session.
+
+   This function uses the STS client to generate a presigned URL for the `get_caller_identity` action, extracts the required values from the URL,
+   and creates a base64-encoded JSON object with these values.
+
+   :param session: A boto3 session object with the necessary AWS credentials and region information.
+   :return: A base64-encoded JSON string containing the required authentication information.
+   :raises ValueError: If the session does not have a region name set.
+   """
+
+    if session.region_name is None:
+        raise ValueError("Region name is not set in the session")
+
+    from botocore.client import Config
+    sts_client = session.client('sts', config=Config(signature_version='v4'))
+    presigned_url = sts_client.generate_presigned_url('get_caller_identity', Params={}, ExpiresIn=3600)
+
+    # Parse the URL to extract components
+    parsed_url = urlparse(presigned_url)
+    query_params = parse_qs(parsed_url.query)
+
+    # Extract values from query parameters
+    json_object = {
+        "method": "POST",
+        "host": parsed_url.hostname,
+        "region": session.region_name,
+        "action": query_params['Action'][0],
+        "date": query_params['X-Amz-Date'][0],
+        "expiration_duration": "3600",  # 1 hour in seconds
```

Comment:
Should be the value in `X-Amz-Expires` and default + overridable by client, because:
1. It's something we support override on the server level In fluffy optional, see [auth.external.aws_auth.get_caller_identity_max_age](https://docs.lakefs.io/enterprise/fluffy-configuration.html). The goal is to have a very short live "identity token". 
2. `3600` is a lie and only works for S3, in reality sts will enforce expiration after `15m`. 
In Java the default is set to `60 seconds`. 
If the fluffy server will configure  anything less than 3600 it will error.

Reply:
Fixed